src/com/mendix/widget/GoogleMaps/components/GoogleMapContainer.ts

```diff
@@ -0,0 +1,171 @@
+            ];
+            attributeList.forEach((attr) =>
+                this.subscriptionHandles.push(window.mx.data.subscribe({
+                    attr,
+                    callback: () => this.fetchData(),
+                    guid: contextObject.getGuid()
+                }))
+            );
+        }
+    }
+
+    private fetchData() {
+        if (this.props.dataSource === "static") {
+            this.setState({ locations: this.props.staticLocations });
+        } else if (this.props.dataSource === "context" && this.props.locationsEntity) {
+            this.fetchLocationsByContext(this.props.contextObject);
+        } else if (this.props.dataSource === "XPath" && this.props.locationsEntity) {
+            this.fetchLocationsByXPath(this.props.contextObject ? this.props.contextObject.getGuid() : "");
+        } else if (this.props.dataSource === "microflow" && this.props.dataSourceMicroflow) {
+            this.fetchLocationsByMicroflow(this.props.dataSourceMicroflow);
+        }
+    }
+
+    private fetchLocationsByContext(contextObject?: mendix.lib.MxObject) {
+        if (contextObject) {
+            this.setLocationsFromMxObjects([ contextObject ]);
+        }
+    }
+
+    private fetchLocationsByXPath(contextGuid: string) {
+        const { entityConstraint } = this.props;
+        const requiresContext = entityConstraint && entityConstraint.indexOf("[%CurrentObject%]") > -1;
+        if (!contextGuid && requiresContext) {
+            this.setState({ locations: [] });
+            return;
+        }
+
+        const constraint = entityConstraint ? entityConstraint.replace("[%CurrentObject%]", contextGuid) : "";
+        window.mx.data.get({
+            callback: mxObjects => this.setLocationsFromMxObjects(mxObjects),
+            error: error =>
+                this.setState({
+                    alertMessage: `An error occurred while retrieving items: ${error}`,
+                    locations: []
+                }),
+            xpath: `//${this.props.locationsEntity}${constraint}`
+        });
+    }
+
+    private fetchLocationsByMicroflow(microflow: string) {
+        if (microflow) {
+            window.mx.ui.action(microflow, {
+                callback: (mxObjects: mendix.lib.MxObject[]) => this.setLocationsFromMxObjects(mxObjects),
+                error: error =>
+                    this.setState({
+                        alertMessage: `An error occurred while retrieving locations: ${error.message}`,
+                        locations: []
+                    }),
+                params: {
+                    guids: this.props.contextObject ? [ this.props.contextObject.getGuid() ] : []
+                }
+            });
+        }
+    }
+
+    private setLocationsFromMxObjects(mxObjects: mendix.lib.MxObject[]): void {
+        const locations = mxObjects.map((mxObject) => ({
+            address: mxObject ? mxObject.get(this.props.addressAttribute) as string : undefined,
```

Comment:
Address attribute is still required. Its up to the user to decide if they are using it or not. in the code what i have done is that if the coordinates are available i use them instead of the address. If both the address and coordinates are provided then i take the coordinates instead.